sdk-commons/src/main/java/org/openmrs/maven/plugins/utility/DefaultWizard.java

```diff
@@ -413,15 +434,32 @@ public String promptForExistingServerIdIfMissing(String serverId) throws MojoExe
 	}
 
 	@Override
-	public String promptForPlatformVersionIfMissing(String version, List<String> versions) throws MojoExecutionException {
-		return promptForMissingValueWithOptions(PLATFORM_VERSION_PROMPT,
-				version, "version", versions, "Please specify platform version", null);
+	public String promptForPlatformVersion(VersionsHelper versionsHelper) throws MojoExecutionException {
+		return promptForPlatformArtifact(versionsHelper).getVersion();
+	}
+
+	@Override
+	public Artifact promptForPlatformArtifact(VersionsHelper versionsHelper) throws MojoExecutionException {
+		Artifact artifact = new Artifact(SDKConstants.PLATFORM_ARTIFACT_ID, SDKConstants.SETUP_DEFAULT_PLATFORM_VERSION, Artifact.GROUP_DISTRO);
+		String version = promptForArtifactVersion( "Please specify platform version", artifact, "Please specify platform version", versionsHelper);
+		artifact.setVersion(version);
+		return artifact;
+	}
+
+	@Override
+	public Artifact promptForRefApp2xArtifact(VersionsHelper versionsHelper) throws MojoExecutionException {
+		Artifact artifact = new Artifact(REFAPP_2X_ARTIFACT_ID, "2.3.1", REFAPP_2X_GROUP_ID, REFAPP_2X_TYPE);
+		String version = promptForArtifactVersion("Please choose a " + REFAPP_2X_PROMPT + " version", artifact, null, versionsHelper);
+		artifact.setVersion(version);
+		return artifact;
 	}
 
 	@Override
-	public String promptForPlatformVersion(List<String> versions) throws MojoExecutionException {
-		return promptForMissingValueWithOptions(PLATFORM_VERSION_PROMPT,
-				null, "version", versions, "Please specify platform version", null);
+	public Artifact promptForRefApp3xArtifact(VersionsHelper versionsHelper) throws MojoExecutionException {
+		Artifact artifact = new Artifact(REFAPP_3X_ARTIFACT_ID, "3.0.0", REFAPP_3X_GROUP_ID, REFAPP_3X_TYPE);
+		String version = promptForArtifactVersion("Please choose a " + REFAPP_3X_PROMPT + " version", artifact, null, versionsHelper);
+		artifact.setVersion(version);
+		return artifact;
```

Comment:
Much more straightforward and simplified methods here for prompting for versions, and able to remove the web of other methods below that are no longer used or needed.